### ISSUE TITLE
Feat: Prometheus input

### ISSUE THREAD

**Issue by georgettica** [salesforce/Merlion#11]
It seems like this repo can be very useful in reading timeseries data and creating projections..

What about adding Prometheus as an input source?
See a middleware layer you can use https://pypi.org/project/prometheus-pandas/

**Comment by paulkass** [comment]
Prometheus seems like an interesting integration to have. The only thing though is that it seems like a lot of work:

* The middleware layer you suggested is labeled explicitly as "intended for use in Jupyter notebooks". We'll need to make sure that it scales well in production environments. If it doesn't, we'll need to fallback to the low-level API which would take even more work.
* We'll need to create an authentication handler in a secure manner. Prometheus is a standalone queryable service with potentially very valuable data, so the connection layer (which we don't have implemented with any other dataset) will be need to be extremely secure.

So basically if the amount of usage this feature will get can vindicate ^ then I think this would be a great addition to the project.

At this time, your best bet is probably to download the prometheus dataset separately and then creating your own dataset as in the examples from those files stored locally.
